File: 10.0.3.020/examples/java/examples/vslcopystream.java
/*******************************************************************************
!                             INTEL CONFIDENTIAL
!   Copyright(C) 2007-2008 Intel Corporation. All Rights Reserved.
!   The source code contained  or  described herein and all documents related to
!   the source code ("Material") are owned by Intel Corporation or its suppliers
!   or licensors.  Title to the  Material remains with  Intel Corporation or its
!   suppliers and licensors. The Material contains trade secrets and proprietary
!   and  confidential  information of  Intel or its suppliers and licensors. The
!   Material  is  protected  by  worldwide  copyright  and trade secret laws and
!   treaty  provisions. No part of the Material may be used, copied, reproduced,
!   modified, published, uploaded, posted, transmitted, distributed or disclosed
!   in any way without Intel's prior express written permission.
!   No license  under any  patent, copyright, trade secret or other intellectual
!   property right is granted to or conferred upon you by disclosure or delivery
!   of the Materials,  either expressly, by implication, inducement, estoppel or
!   otherwise.  Any  license  under  such  intellectual property  rights must be
!   express and approved by Intel in writing.
!******************************************************************************/

import com.intel.mkl.*;

public final class vslcopystream {
    /**
     * Incarnation prohibited.
     */
    private vslcopystream() {}

    /**
     * Individual name of this test.
     */
    private final static String test_name = "vslcopystream";

    /**
     * No command-line options.
     */
    public static void main(String[] args) {
        try {
            test();
        } catch (Exception statusException) {
            statusException.printStackTrace(System.out);
            System.out.println("\nTEST FAILED");
            System.exit(1);
        }
        System.out.println("\nTEST PASSED");
    }

    /**
     * Demonstrate the VSL functions and test them.
     */
    public static void test() throws Exception
    {
		// Parameters
        final int SEED = 7373737;
        final int BRNG = VSL.BRNG_MCG31;
        final int METHOD = 0;
        final int N = 1000;

		int r[] = new int [N];
		int rCpy[] = new int [N];
		int i, err = 0;

		/***** Initialize streams *****/
		VSLStreamStatePtr stream = new VSLStreamStatePtr();
		int errcode = VSL.vslNewStream(stream,BRNG, SEED);
		VSLStreamStatePtr streamCpy = new VSLStreamStatePtr();
		errcode = VSL.vslCopyStream( streamCpy, stream );
		ErrCheck.CheckVslError( errcode );

		/***** Call RNGs *****/
		errcode = VSL.viRngUniformBits( METHOD, stream,   N, r );
		ErrCheck.CheckVslError( errcode );
		errcode = VSL.viRngUniformBits( METHOD, streamCpy,   N, rCpy );
		ErrCheck.CheckVslError( errcode );

		/***** Compare results *****/
		for(i = 0; i < N; i++){
			if(r[i] != rCpy[i]) err++;
		}

		/***** Printing results *****/
		System.out.println("");
		System.out.println(" Sample of vslCopyStream");
		System.out.println(" ------------------------");
		System.out.println("");
		System.out.println(" Parameters:");
		System.out.println("    seed   =   " + SEED);


		System.out.println(" Results (first 10 of 1000):");
		System.out.println(" ---------------------------");
		for(i=0;i<10;i++) {
			System.out.println("r[" + i + "]=" + r[i] + " rCpy[" + i + "]=" + rCpy[i]);
		}

		System.out.println("");
		if(err > 0) {
			System.out.println("Error: " + err + " values are incorrect!");
		} else {
			System.out.println(" Results of original stream and its copy are identical.");
		}

		System.out.println("");

		/***** Deinitialize *****/
		errcode = VSL.vslDeleteStream( stream );
		ErrCheck.CheckVslError( errcode );
		errcode = VSL.vslDeleteStream( streamCpy );
		ErrCheck.CheckVslError( errcode );

	}
}

File: 10.0.3.020/examples/vslf/source/vslcopystream.f
!*******************************************************************************
!                             INTEL CONFIDENTIAL
!  Copyright(C) 2003-2008 Intel Corporation. All Rights Reserved.
!  The source code contained  or  described herein and all documents related to
!  the source code ("Material") are owned by Intel Corporation or its suppliers
!  or licensors.  Title to the  Material remains with  Intel Corporation or its
!  suppliers and licensors. The Material contains trade secrets and proprietary
!  and  confidential  information of  Intel or its suppliers and licensors. The
!  Material  is  protected  by  worldwide  copyright  and trade secret laws and
!  treaty  provisions. No part of the Material may be used, copied, reproduced,
!  modified, published, uploaded, posted, transmitted, distributed or disclosed
!  in any way without Intel's prior express written permission.
!  No license  under any  patent, copyright, trade secret or other intellectual
!  property right is granted to or conferred upon you by disclosure or delivery
!  of the Materials,  either expressly, by implication, inducement, estoppel or
!  otherwise.  Any  license  under  such  intellectual property  rights must be
!  express and approved by Intel in writing.
!
!*******************************************************************************
!  Content:
!    vslCopyStream  Example Program Text
!*******************************************************************************

      include 'mkl_vsl.fi'
      include "errcheck.inc"

      program MKL_VSL_TEST

      USE MKL_VSL_TYPE
      USE MKL_VSL

      TYPE (VSL_STREAM_STATE) :: stream
      TYPE (VSL_STREAM_STATE) :: streamCpy
      integer(kind=4) r(1000)
      integer(kind=4) rCpy(1000)
      integer(kind=4) i,err,nn
      integer n
      integer brng,method,seed
      integer(kind=4) errcode

      err=0
      n=1000
      nn=10
      brng=VSL_BRNG_MCG31
      method=0

!     ***** Initialize seeds *****
      seed=7373737

!     ***** Initialize streams *****
      errcode=vslnewstream  ( stream,   brng,  seed )
      call CheckVslError(errcode)
      errcode=vslcopystream ( streamCpy, stream )
      call CheckVslError(errcode)

!     ***** Call RNGs *****
      errcode=virnguniformbits( method, stream,   n, r )
      call CheckVslError(errcode)
      errcode=virnguniformbits( method, streamCpy, n, rCpy )
      call CheckVslError(errcode)

!     ***** Compare results *****
      do i=1,1000
        if (r(i) .NE. rCpy(i)) then
          err=err+1
        end if
      end do

!     ***** Printing results *****
      print *,"Sample of vslCopyStream"
      print *,"------------------------"
      print *,""
      print *,"Parameters:"
      print 11,"    seed   =  ",seed

      print *,""
      print *,"Results (first 10 of 1000):"
      print *,"---------------------------"
      do i=1,nn
        print 10, "r[",i-1,"]=0x",r(i)," rCpy[",i-1,"]=0x",rCpy(i)
      end do

      print *,""
      if (err>0) then
        print 13,"Error: ", err," values are incorrect!"
        stop 1
      else
        print *,"Results of original stream and its copy are identical."
      end if

!     ***** Deinitialize *****
      errcode=vsldeletestream( stream )
      call CheckVslError(errcode)
      errcode=vsldeletestream( streamCpy )
      call CheckVslError(errcode)

10    format(A,I1,A,Z08,A,I1,A,Z08)
11    format(A,I8)
13    format(A,I8,A)

      end
